Bento/Renderable/Deletable.swift

```diff
@@ -1,15 +1,39 @@
 import UIKit
 
-public protocol Deletable {
-    var canBeDeleted: Bool { get }
+protocol Deletable {
     var deleteActionText: String { get }
     var backgroundColor: UIColor? { get }
 
     func delete()
 }
 
-public extension Deletable {
-    var backgroundColor: UIColor? {
-        return nil
+final class DeletableComponent<Base: Renderable>: AnyRenderableBox<Base>, Deletable where Base.View: UIView  {
+    let deleteActionText: String
+    let backgroundColor: UIColor?
+    private let source: AnyRenderableBox<Base>
```

Comment:
You have the real `Base` here. Why is it necessary to type erase it?

Reply:
Because `Base` does not know how to search for behaviour. Essentially the composition of behaviours is like a liked list, when we ask to `cast(to: )` we first check if type is the one we are implementing, if not we propagate search up in composition hierarchy. For more details please see line 29 in this file